bot/exts/moderation/incidents.py

```diff
@@ -130,6 +138,60 @@ def has_signals(message: discord.Message) -> bool:
     return ALL_SIGNALS.issubset(own_reactions(message))
 
 
+def shorten_text(text: str) -> str:
+    """Truncate the text if there are over 3 lines or 300 characters, or if it is a single word."""
+    original_length = len(text)
+    lines = text.count("\n")
+    # Limit to a maximum of three lines
+    if lines > 3:
+        text = "\n".join(line for line in text.split('\n')[:3])
+    # If it is a single word, then truncate it to 50 characters
+    if text.count(" ") < 1:
+        text = text[:50]
+    # Truncate text to a maximum of 300 characters
+    if len(text) > 300:
+        text = text[:300]
+    # Add placeholder if the text was shortened
+    if len(text) < original_length:
+        text += "..."
+
+    return text
+
+
+async def make_message_link_embed(ctx: Context, message_link: str) -> t.Optional[discord.Embed]:
+    """
+    Create an embedded representation of the discord message link contained in the incident report.
+
+    The Embed would contain the following information -->
+        Author: @Jason Terror ♦ (736234578745884682)
+        Channel: Special/#bot-commands (814190307980607493)
+        Content: This is a very important message!
+    """
+    try:
+        message: discord.Message = await MessageConverter().convert(ctx, message_link)
+
+    except discord.DiscordException as e:
+        log.exception(f"Failed to make message link embed for '{message_link}', raised exception: {e}")
+
```

Comment:
I think we should specifically handle a few cases here that have been missed:
- if a message has been deleted, the converter will raise NotFound. We should have a special message for this case
- When it's deleted, we can search through #message-change-log in our message cache looking for the appropriate message id: {id} string, to link there and extract content from there instead.
- since incidents is within helper view, we should at the very least restrict this converter from accessing messages inside, say, #admins with some kind of a channel check.

Reply:
Mentioned in ed30b1e and 94bcc93.